packages/component-base/src/resizable-mixin.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * @license
+ * Copyright (c) 2021 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { dedupingMixin } from '@polymer/polymer/lib/utils/mixin.js';
+
+/**
+ * A mixin that uses a ResizeObserver to listen to host size changes.
+ *
+ * @polymerMixin
+ */
+export const ResizableMixin = dedupingMixin(
+  (superclass) =>
+    class ResizableMixinClass extends superclass {
+      /** @protected */
+      ready() {
+        super.ready();
+
+        this.__resizeObserver = new ResizeObserver(() => {
```

Comment:
I think we could reuse the same observer for all elements. Since we always observe `this`, and call `this._onResize()`, we should be able to just iterate over all entries of the observer callback and call `entry._onResize()` instead.

Reply:
I considered that but it would require toggling observe/unobserve on connect/disconnect. What do you see as the main benefit in using just one observer instance?

Reply:
Mainly for performance reasons. I would have guessed that mainly just memory usage increases, but it seems that it's also (much) faster if a resize would trigger multiple observers:
- https://github.com/WICG/resize-observer/issues/59
- https://groups.google.com/a/chromium.org/g/blink-dev/c/z6ienONUb5A/m/F5-VcUZtBAAJ

Reply:
Done